pkg/action/pull.go

```diff
@@ -92,13 +92,13 @@ func (p *Pull) Run(chartRef string) (string, error) {
 	}
 
 	if registry.IsOCI(chartRef) {
-		if p.Version == "" {
-			return out.String(), errors.Errorf("--version flag is explicitly required for OCI registries")
-		}
-
 		c.Options = append(c.Options,
-			getter.WithRegistryClient(p.cfg.RegistryClient),
-			getter.WithTagName(p.Version))
+			getter.WithRegistryClient(p.cfg.RegistryClient))
+
+		if p.Version != "" {
+			c.Options = append(c.Options,
+				getter.WithTagName(p.Version))
+		}
```

Comment:
The version here isn't a version but can be a specific version or range. In either case, it's not a tag. So, using `WithTagName` don't convey what it does. As an external API to API users this is confusing.

Also, the other built-in getters don't turn a version range to a version in this part of the code. Doing it here makes it different from the others. I would pull the version part from here. See my other comments on where to put the version handling.

Reply:
done